src/Servers/Kestrel/Core/src/Internal/Http3/Http3ControlStream.cs

```diff
@@ -250,6 +218,33 @@ private async Task HandleControlStream()
             }
         }
 
+        private async ValueTask HandleEncodingTask()
+        {
+            // Noop encoding task. Settings make it so we don't need to read content of encoder and decoder.
+            // An endpoint MUST allow its peer to create an encoder stream and a
+            // decoder stream even if the connection's settings prevent their use.
+
+            while (_isClosed == 0)
```

Comment:
Nit: I think it's better to check result.IsCompleted than _isClosed to avoid spinning if the two ever get out -of-sync (even if they're synchronized today).